2015-09-03  Joseph Myers  <joseph@codesourcery.com>

	* Makefile (headers): Remove bits/libc-lock.h.
	* libio/Makefile (headers): Remove bits/stdio-lock.h.

	* libio/libio.h [_IO_MTSAFE_IO]: Remove include of
	<bits/stdio-lock.h> and commented-out include of <comthread.h>.
	* include/libio.h [!_ISOMAC && _IO_MTSAFE_IO]: Include
	<bits/stdio-lock.h>.
	* stdio-common/scanf15.c (_IO_MTSAFE_IO): Undefine.
	* stdio-common/scanf17.c (_IO_MTSAFE_IO): Likewise.

--- a/Makefile
+++ b/Makefile
@@ -58,7 +58,7 @@ endif # $(AUTOCONF) = no
 		   subdir_objs subdir_stubs subdir_testclean		\
 		   $(addprefix install-, no-libc.a bin lib data headers others)
 
-headers := limits.h values.h features.h gnu-versions.h bits/libc-lock.h \
+headers := limits.h values.h features.h gnu-versions.h \
 	   bits/xopen_lim.h gnu/libc-version.h stdc-predef.h
 
 echo-headers: subdir_echo-headers
--- a/include/libio.h
+++ b/include/libio.h
@@ -1,3 +1,6 @@
+#if !defined _ISOMAC && defined _IO_MTSAFE_IO
+# include <bits/stdio-lock.h>
+#endif
 #include <libio/libio.h>
 
 #ifndef _ISOMAC
--- a/libio/Makefile
+++ b/libio/Makefile
@@ -22,7 +22,7 @@ subdir	:= libio
 
 include ../Makeconfig
 
-headers	:= stdio.h libio.h _G_config.h bits/stdio.h bits/stdio-lock.h \
+headers	:= stdio.h libio.h _G_config.h bits/stdio.h \
 	   bits/sys_errlist.h bits/stdio2.h bits/stdio-ldbl.h bits/libio-ldbl.h
 
 routines	:=							      \
--- a/libio/libio.h
+++ b/libio/libio.h
@@ -145,11 +145,7 @@ struct _IO_jump_t;  struct _IO_FILE;
 
 /* Handle lock.  */
 #ifdef _IO_MTSAFE_IO
-# if defined __GLIBC__ && __GLIBC__ >= 2
-#  include <bits/stdio-lock.h>
-# else
-/*# include <comthread.h>*/
-# endif
+/* _IO_lock_t defined in internal headers during the glibc build.  */
 #else
 typedef void _IO_lock_t;
 #endif
--- a/stdio-common/scanf15.c
+++ b/stdio-common/scanf15.c
@@ -1,6 +1,7 @@
 #undef _GNU_SOURCE
 #define _XOPEN_SOURCE 600
 #undef _LIBC
+#undef _IO_MTSAFE_IO
 /* The following macro definitions are a hack.  They word around disabling
    the GNU extension while still using a few internal headers.  */
 #define u_char unsigned char
--- a/stdio-common/scanf17.c
+++ b/stdio-common/scanf17.c
@@ -1,6 +1,7 @@
 #undef _GNU_SOURCE
 #define _XOPEN_SOURCE 600
 #undef _LIBC
+#undef _IO_MTSAFE_IO
 /* The following macro definitions are a hack.  They word around disabling
    the GNU extension while still using a few internal headers.  */
 #define u_char unsigned char
